tests/phpunit/CivibuildGeneralTest.php

```diff
@@ -0,0 +1,164 @@
+  }
+
+  public function testList_Create_List_Destroy_List() {
+    $result = ProcessUtil::runOk($this->cmd("civibuild list"));
+    $this->assertNotContains($this->buildName, $result->getOutput());
+
+    $result = ProcessUtil::runOk($this->cmd(
+      "civibuild create {$this->buildName} --type $this->buildType"
+    ));
+    $this->assertRegExp("; - CMS_ROOT: [^\n]*/{$this->buildName};", $result->getOutput());
+
+    $result = ProcessUtil::runOk($this->cmd("civibuild list"));
+    $this->assertContains($this->buildName, $result->getOutput());
+
+    $result = ProcessUtil::runOk($this->cmd(
+      "echo y | civibuild destroy {$this->buildName}"
+    ));
+    $this->assertFalse(file_exists($this->getAbsPath($this->buildName)));
+    $this->assertFalse(file_exists($this->getAbsPath($this->buildName . '.sh')));
+
+    $result = ProcessUtil::runOk($this->cmd("civibuild list"));
+    $this->assertNotContains($this->buildName, $result->getOutput());
+  }
+
+  public function getCommandsWhichSupportDefaultBuild() {
+    return array(
+      array('reinstall'),
+      array('snapshot'),
+      array('restore'),
+      array('show'),
+    );
+  }
+
+  /**
+   * @param $command
+   * @dataProvider getCommandsWhichSupportDefaultBuild
+   */
+  public function testUseDefaultBuildName($command) {
+    $result = ProcessUtil::runOk($this->cmd(
+      "civibuild create {$this->buildName} --type $this->buildType"
+    ));
+    $this->assertRegExp("; - CMS_ROOT: [^\n]*/{$this->buildName};", $result->getOutput());
+    $this->assertRegExp('; - CIVI_DB_DSN: mysql://.*:.*@.*/.*;', $result->getOutput());
+    $this->assertTrue(file_exists($this->getAbsPath($this->buildName)));
+    $this->assertTrue(file_exists($this->getAbsPath($this->buildName . '.sh')));
+
+    $buildDir = $this->getAbsPath($this->buildName);
+    mkdir("$buildDir/subdir");
+
+    // Run the command three different ways, and see if the results are the same.
+
+    $resultWithArg = ProcessUtil::runOk($this->cmd("civibuild $command {$this->buildName}"));
+    $this->assertNotEmpty($resultWithArg->getOutput());
+
+    $resultInSubdir = ProcessUtil::runOk(
+      $this->cmd("civibuild $command")->setWorkingDirectory("$buildDir/subdir")
+    );
+
+    $this->assertContains("[[Detected SITE_NAME: {$this->buildName}]]\n", $resultInSubdir->getOutput());
+    $this->assertEquals(
+      $resultWithArg->getOutput(),
+      str_replace("[[Detected SITE_NAME: {$this->buildName}]]\n", "", $resultInSubdir->getOutput())
+    );
+
+    // FIXME: Bug!
+    //    $resultInDir = ProcessUtil::runOk(
+    //      $this->cmd("civibuild $command")->setWorkingDirectory($buildDir)
+    //    );
+    //    $this->assertContains("[[Detected SITE_NAME: {$this->buildName}]]\n", $resultInSubdir->getOutput());
+    //    $this->assertEquals(
+    //      $resultWithArg->getOutput(),
+    //      str_replace("[[Detected SITE_NAME: {$this->buildName}]]\n", "", $resultInDir->getOutput())
+    //    );
```

Comment:
@ginkgomzd, while adding unit tests for civibuild, I bumped into an edge-case where `civibuild_detect_site_name` stumbles -- it works correctly in subdirs (e.g. `build/dmaster/sites/all/modules`) but not in root dirs (e.g. `build/dmaster`). The example above demonstrates this.